src/olympia/shelves/serializers.py

```diff
@@ -0,0 +1,26 @@
+from rest_framework import serializers
+from rest_framework.reverse import reverse as drf_reverse
+
+from django.conf import settings
+
+from olympia.shelves.models import Shelf
+
+
+class ShelfSerializer(serializers.ModelSerializer):
+    url = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Shelf
+        fields = ['title', 'url', 'footer_text', 'footer_pathname']
+
+    def get_url(self, obj):
+        baseUrl = settings.INTERNAL_SITE_URL
+        if obj.endpoint == 'search':
+            api = drf_reverse('v4:addon-search')
```

Comment:
because this is in the API we want to always use the matching API version in the reverses too.  So strip off the `v4:` at the start of the url name and include `request=self.context.get('request')` as a kwarg to pass down the request so the api version can be resolved from that by DRF.